ui/pages/confirmations/components/simulation-details/fiat-display.test.tsx

```diff
@@ -10,38 +11,70 @@ import { FIAT_UNAVAILABLE } from './types';
 const store = configureStore()(mockState);
 
 jest.mock('../../../../hooks/useFiatFormatter');
-(useFiatFormatter as jest.Mock).mockReturnValue((value: number) => `$${value}`);
-
-describe('IndividualFiatDisplay', () => {
-  // @ts-expect-error This is missing from the Mocha type definitions
-  it.each([
-    [FIAT_UNAVAILABLE, 'Not Available'],
-    [100, '$100'],
-    [-100, '$100'],
-  ])(
-    'when fiatAmount is %s it renders %s',
-    (fiatAmount: number | null, expected: string) => {
-      renderWithProvider(
-        <IndividualFiatDisplay fiatAmount={fiatAmount} />,
+jest.mock('../../../../hooks/useHideFiatForTestnet');
+
+describe('FiatDisplay', () => {
+  const mockUseHideFiatForTestnet = jest.mocked(useHideFiatForTestnet);
+  const mockUseFiatFormatter = jest.mocked(useFiatFormatter);
+
+  beforeEach(() => {
+    jest.resetAllMocks();
+    mockUseHideFiatForTestnet.mockReturnValue(false);
```

Comment:
Since we're already using `renderWithProvider`, should we instead be mocking the preference state as opposed to mocking selectors as recommended by Redux?

Reply:
For sure, changed approach in https://github.com/MetaMask/metamask-extension/pull/25167/commits/07411341fe588535c04301befa1bc358135e0ced